Marlin/SanityCheck.h

```diff
@@ -441,12 +441,8 @@
 /**
  * Allen Key Z probe requires Delta and Auto Bed Leveling grid
  */
-#if ENABLED(Z_PROBE_ALLEN_KEY)
-  #if !ENABLED(DELTA)
-    #error "Z_PROBE_ALLEN_KEY is only usable with DELTA."
-  #elif ENABLED(MESH_BED_LEVELING) || (ENABLED(AUTO_BED_LEVELING_FEATURE) && !ENABLED(AUTO_BED_LEVELING_GRID))
-    #error "Z_PROBE_ALLEN_KEY can only use AUTO_BED_LEVELING_GRID leveling."
-  #endif
+#if ENABLED(Z_PROBE_ALLEN_KEY) && (Z_HOME_DIR < 0) && ENABLED(Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN) && DISABLED(TEST_ALLEN_KEY)
+  #error "You can't home to a z min endstop with a Z_PROBE_ALLEN_KEY"
```

Comment:
This is not much more dangerous than doing the same with a servo probe. But i'm scared by the much larger (unhomed) z-moves.